components/server/src/prebuilds/prebuild-manager.ts

```diff
@@ -416,6 +416,7 @@ export class PrebuildManager {
                     true,
                 );
                 if (prebuild) {
+                    // TODO(gpl): Why not "done: prebuild.state === "available""?
```

Comment:
Yea, why not? It would make sense if we called `findBaseForIncrementalWorkspace` with `includeUnfinishedPrebuilds: false` here, but we don't. Perhaps an oversight in https://github.com/gitpod-io/gitpod/pull/14461?

We have the same assumption just above here: https://github.com/gitpod-io/gitpod/blob/58ca82104164b1317db0ffd9774906bb50b5344d/components/server/src/prebuilds/prebuild-manager.ts#L379-L385

Digging deeper to the places that call this function, it looks like we actually depend on this behavior when reporting if we've started a prebuild when handling a webhook event. There are some more quirks (like GitLab reporting that unconditionally), but maybe it's some ancient way of communicating not a `done` status, but a `triggered` one? 😆 
https://github.com/gitpod-io/gitpod/blob/58ca82104164b1317db0ffd9774906bb50b5344d/components/server/src/prebuilds/bitbucket-server-app.ts#L188-L194

Reply:
> but maybe it's some ancient way of communicating not a done status, but a triggered one? 😆

`done` - at least originally, when introduced in 2018/2019 - meant "the prebuild is done (aka available)". :smile: 

> There are some more quirks (like GitLab reporting that unconditionally)

What exactly do you mean? Somehow this gives me the feeling you found the root cause of the bug we are after...? :eyes:

Reply:
> when starting a prebuild on when responding to webhoooks

Ah, gotcha. Yes, oversights from the PR you cited. :+1:

Reply:
Pushed the `prebuild.state === "available"` change.